cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -164,18 +163,26 @@ static Status AppendObjectBinaries(PyArrayObject* arr, PyArrayObject* mask,
     if ((have_mask && mask_values[offset]) || PandasObjectIsNull(obj)) {
       RETURN_NOT_OK(builder->AppendNull());
       continue;
-    } else if (!PyBytes_Check(obj)) {
+    } else if (PyBytes_Check(obj)) {
+      const int32_t length = static_cast<int32_t>(PyBytes_GET_SIZE(obj));
+      if (ARROW_PREDICT_FALSE(builder->value_data_length() + length >
+                              kBinaryMemoryLimit)) {
+        break;
```

Comment:
I have trouble parsing this... If we don't know how to convert this, we should fail, not simply break from the loop, no?

Reply:
After reading the code a bit more carefully, I understand... though there is still a problem: what if `length` is larger than `kBinaryMemoryLimit`?